diff -ruN a/app/Actions/Fortify/CreateNewUser.php b/app/Actions/Fortify/CreateNewUser.php
--- a/app/Actions/Fortify/CreateNewUser.php	1969-12-31 18:00:00
+++ b/app/Actions/Fortify/CreateNewUser.php	2026-07-17 23:35:13
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Fortify;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+use Laravel\Fortify\Contracts\CreatesNewUsers;
+
+class CreateNewUser implements CreatesNewUsers
+{
+    use PasswordValidationRules;
+
+    /**
+     * @param array<string, string> $input
+     */
+    public function create(array $input): User
+    {
+        Validator::make($input, [
+            'email' => [
+                'required',
+                'string',
+                'max:255',
+                'email:strict,dns',
+                'indisposable',
+                Rule::unique(User::class),
+            ],
+            'first_name' => ['required', 'string', 'max:255'],
+            'last_name' => ['required', 'string', 'max:255'],
+            'password' => $this->passwordRules(),
+        ])->validate();
+
+        return User::query()->create([
+            'email' => $input['email'],
+            'first_name' => $input['first_name'],
+            'last_name' => $input['last_name'],
+            'password' => Hash::make($input['password']),
+        ]);
+    }
+}
diff -ruN a/app/Actions/Fortify/PasswordValidationRules.php b/app/Actions/Fortify/PasswordValidationRules.php
--- a/app/Actions/Fortify/PasswordValidationRules.php	1969-12-31 18:00:00
+++ b/app/Actions/Fortify/PasswordValidationRules.php	2026-07-17 23:35:13
@@ -0,0 +1,19 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Fortify;
+
+use Illuminate\Contracts\Validation\Rule;
+use Illuminate\Validation\Rules\Password;
+
+trait PasswordValidationRules
+{
+    /**
+     * @return array<int, array<mixed>|Rule|string>
+     */
+    protected function passwordRules(): array
+    {
+        return ['required', 'string', Password::default(), 'confirmed'];
+    }
+}
diff -ruN a/app/Actions/Fortify/ResetUserPassword.php b/app/Actions/Fortify/ResetUserPassword.php
--- a/app/Actions/Fortify/ResetUserPassword.php	1969-12-31 18:00:00
+++ b/app/Actions/Fortify/ResetUserPassword.php	2026-07-17 23:35:13
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Fortify;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Laravel\Fortify\Contracts\ResetsUserPasswords;
+
+class ResetUserPassword implements ResetsUserPasswords
+{
+    use PasswordValidationRules;
+
+    /**
+     * @param array<string, string> $input
+     */
+    public function reset(User $user, array $input): void
+    {
+        Validator::make($input, [
+            'password' => $this->passwordRules(),
+        ])->validate();
+
+        $user->forceFill([
+            'password' => Hash::make($input['password']),
+        ])->save();
+    }
+}
diff -ruN a/app/Actions/Fortify/UpdateUserPassword.php b/app/Actions/Fortify/UpdateUserPassword.php
--- a/app/Actions/Fortify/UpdateUserPassword.php	1969-12-31 18:00:00
+++ b/app/Actions/Fortify/UpdateUserPassword.php	2026-07-17 23:35:13
@@ -0,0 +1,30 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Fortify;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Facades\Validator;
+use Laravel\Fortify\Contracts\UpdatesUserPasswords;
+
+class UpdateUserPassword implements UpdatesUserPasswords
+{
+    use PasswordValidationRules;
+
+    /**
+     * @param array<string, string> $input
+     */
+    public function update(User $user, array $input): void
+    {
+        Validator::make($input, [
+            'current_password' => ['required', 'string', 'current_password:web'],
+            'password' => $this->passwordRules(),
+        ])->validateWithBag('updatePassword');
+
+        $user->forceFill([
+            'password' => Hash::make($input['password']),
+        ])->save();
+    }
+}
diff -ruN a/app/Actions/Fortify/UpdateUserProfileInformation.php b/app/Actions/Fortify/UpdateUserProfileInformation.php
--- a/app/Actions/Fortify/UpdateUserProfileInformation.php	1969-12-31 18:00:00
+++ b/app/Actions/Fortify/UpdateUserProfileInformation.php	2026-07-17 23:35:13
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Fortify;
+
+use App\Models\User;
+use Illuminate\Support\Facades\Validator;
+use Illuminate\Validation\Rule;
+use Laravel\Fortify\Contracts\UpdatesUserProfileInformation;
+
+class UpdateUserProfileInformation implements UpdatesUserProfileInformation
+{
+    /**
+     * @param array{email: string, first_name: string, last_name: string} $input
+     */
+    public function update(User $user, array $input): void
+    {
+        Validator::make($input, [
+            'email' => [
+                'required',
+                'string',
+                'email',
+                'max:255',
+                Rule::unique('users')->ignore($user->id),
+            ],
+            'first_name' => ['required', 'string', 'max:255'],
+            'last_name' => ['required', 'string', 'max:255'],
+        ])->validateWithBag('updateProfileInformation');
+
+        if ($input['email'] !== $user->email) {
+            $this->updateVerifiedUser($user, $input);
+        } else {
+            $user->forceFill([
+                'email' => $input['email'],
+                'first_name' => $input['first_name'],
+                'last_name' => $input['last_name'],
+            ])->save();
+        }
+    }
+
+    /**
+     * @param array{email: string, first_name: string, last_name: string} $input
+     */
+    protected function updateVerifiedUser(User $user, array $input): void
+    {
+        $user->forceFill([
+            'email' => $input['email'],
+            'email_verified_at' => null,
+            'first_name' => $input['first_name'],
+            'last_name' => $input['last_name'],
+        ])->save();
+
+        $user->sendEmailVerificationNotification();
+    }
+}
diff -ruN a/app/Providers/FortifyServiceProvider.php b/app/Providers/FortifyServiceProvider.php
--- a/app/Providers/FortifyServiceProvider.php	1969-12-31 18:00:00
+++ b/app/Providers/FortifyServiceProvider.php	2026-07-18 06:25:21
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Providers;
+
+use App\Actions\Fortify\CreateNewUser;
+use App\Actions\Fortify\ResetUserPassword;
+use App\Actions\Fortify\UpdateUserPassword;
+use App\Actions\Fortify\UpdateUserProfileInformation;
+use Illuminate\Cache\RateLimiting\Limit;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\RateLimiter;
+use Illuminate\Support\ServiceProvider;
+use Illuminate\Support\Str;
+use Laravel\Fortify\Actions\RedirectIfTwoFactorAuthenticatable;
+use Laravel\Fortify\Fortify;
+
+class FortifyServiceProvider extends ServiceProvider
+{
+    public function boot(): void
+    {
+        Fortify::createUsersUsing(CreateNewUser::class);
+        Fortify::redirectUserForTwoFactorAuthenticationUsing(RedirectIfTwoFactorAuthenticatable::class);
+        Fortify::resetUserPasswordsUsing(ResetUserPassword::class);
+        Fortify::updateUserPasswordsUsing(UpdateUserPassword::class);
+        Fortify::updateUserProfileInformationUsing(UpdateUserProfileInformation::class);
+
+        RateLimiter::for('login', function (Request $request) {
+            $throttleKey = Str::transliterate(sprintf(
+                '%s|%s',
+                Str::lower((string) $request->string(Fortify::username())),
+                $request->ip()
+            ));
+
+            return Limit::perMinute(5)->by($throttleKey);
+        });
+
+        RateLimiter::for('two-factor', function (Request $request) {
+            return Limit::perMinute(5)->by($request->session()->get('login.id'));
+        });
+    }
+}
diff -ruN a/boost.json b/boost.json
--- a/boost.json	2026-07-18 00:00:00
+++ b/boost.json	2026-07-18 00:00:00
@@ -14 +14,2 @@
+        "fortify-development",
         "laravel-best-practices",
diff -ruN a/bootstrap/app.php b/bootstrap/app.php
--- a/bootstrap/app.php	2026-07-18 00:00:00
+++ b/bootstrap/app.php	2026-07-18 00:00:00
@@ -16,2 +16,3 @@
         $middleware
+            ->redirectGuestsTo(fn (): string => route('login'))
             ->redirectUsersTo('/')
diff -ruN a/bootstrap/providers.php b/bootstrap/providers.php
--- a/bootstrap/providers.php	2026-07-18 00:00:00
+++ b/bootstrap/providers.php	2026-07-18 00:00:00
@@ -5 +5,2 @@
 use App\Providers\AppServiceProvider;
+use App\Providers\FortifyServiceProvider;
@@ -8 +9,2 @@
     AppServiceProvider::class,
+    FortifyServiceProvider::class,
diff -ruN a/composer.json b/composer.json
--- a/composer.json	2026-07-18 00:00:00
+++ b/composer.json	2026-07-18 00:00:00
@@ -20 +20,2 @@
+    "laravel/fortify": "^1.37.2",
     "laravel/framework": "^13.19.0",
@@ -22 +23,2 @@
+    "propaganistas/laravel-disposable-email": "^2.5.0",
     "propaganistas/laravel-phone": "^6.0.3",
diff -ruN a/config/fortify.php b/config/fortify.php
--- a/config/fortify.php	1969-12-31 18:00:00
+++ b/config/fortify.php	2026-07-17 23:35:13
@@ -0,0 +1,161 @@
+<?php
+
+declare(strict_types=1);
+
+use Laravel\Fortify\Features;
+
+return [
+
+    /*
+    |--------------------------------------------------------------------------
+    | Fortify Guard
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify which authentication guard Fortify will use while
+    | authenticating users. This value should correspond with one of your
+    | guards that is already present in your "auth" configuration file.
+    |
+    */
+
+    'guard' => 'web',
+
+    /*
+    |--------------------------------------------------------------------------
+    | Fortify Password Broker
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify which password broker Fortify can use when a user
+    | is resetting their password. This configured value should match one
+    | of your password brokers setup in your "auth" configuration file.
+    |
+    */
+
+    'passwords' => 'users',
+
+    /*
+    |--------------------------------------------------------------------------
+    | Username / Email
+    |--------------------------------------------------------------------------
+    |
+    | This value defines which model attribute should be considered as your
+    | application's "username" field. Typically, this might be the email
+    | address of the users but you are free to change this value here.
+    |
+    | Out of the box, Fortify expects forgot password and reset password
+    | requests to have a field named 'email'. If the application uses
+    | another name for the field you may define it below as needed.
+    |
+    */
+
+    'username' => 'email',
+
+    'email' => 'email',
+
+    /*
+    |--------------------------------------------------------------------------
+    | Lowercase Usernames
+    |--------------------------------------------------------------------------
+    |
+    | This value defines whether usernames should be lowercased before saving
+    | them in the database, as some database system string fields are case
+    | sensitive. You may disable this for your application if necessary.
+    |
+    */
+
+    'lowercase_usernames' => true,
+
+    /*
+    |--------------------------------------------------------------------------
+    | Home Path
+    |--------------------------------------------------------------------------
+    |
+    | Here you may configure the path where users will get redirected during
+    | authentication or password reset when the operations are successful
+    | and the user is authenticated. You are free to change this value.
+    |
+    */
+
+    'home' => '/',
+
+    /*
+    |--------------------------------------------------------------------------
+    | Fortify Routes Prefix / Subdomain
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify which prefix Fortify will assign to all the routes
+    | that it registers with the application. If necessary, you may change
+    | subdomain under which all of the Fortify routes will be available.
+    |
+    */
+
+    'prefix' => '',
+
+    'domain' => null,
+
+    /*
+    |--------------------------------------------------------------------------
+    | Fortify Routes Middleware
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify which middleware Fortify will assign to the routes
+    | that it registers with the application. If necessary, you may change
+    | these middleware but typically this provided default is preferred.
+    |
+    */
+
+    'middleware' => ['web'],
+
+    /*
+    |--------------------------------------------------------------------------
+    | Rate Limiting
+    |--------------------------------------------------------------------------
+    |
+    | By default, Fortify will throttle logins to five requests per minute for
+    | every email and IP address combination. However, if you would like to
+    | specify a custom rate limiter to call then you may specify it here.
+    |
+    */
+
+    'limiters' => [
+        'login' => 'login',
+        'two-factor' => 'two-factor',
+    ],
+
+    /*
+    |--------------------------------------------------------------------------
+    | Register View Routes
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify if the routes returning views should be disabled as
+    | you may not need them when building your own application. This may be
+    | especially true if you're writing a custom single-page application.
+    |
+    */
+
+    'views' => false,
+
+    /*
+    |--------------------------------------------------------------------------
+    | Features
+    |--------------------------------------------------------------------------
+    |
+    | Some of the Fortify features are optional. You may disable the features
+    | by removing them from this array. You're free to only remove some of
+    | these features or you can even remove all of these if you need to.
+    |
+    */
+
+    'features' => [
+        // Features::emailVerification(),
+        Features::registration(),
+        // Features::resetPasswords(),
+        // Features::twoFactorAuthentication([
+        //     'confirm' => true,
+        //     'confirmPassword' => true,
+        //     // 'window' => 0,
+        // ]),
+        // Features::updatePasswords(),
+        // Features::updateProfileInformation(),
+    ],
+
+];
diff -ruN a/database/migrations/2025_08_31_000154_add_two_factor_columns_to_users_table.php b/database/migrations/2025_08_31_000154_add_two_factor_columns_to_users_table.php
--- a/database/migrations/2025_08_31_000154_add_two_factor_columns_to_users_table.php	1969-12-31 18:00:00
+++ b/database/migrations/2025_08_31_000154_add_two_factor_columns_to_users_table.php	2026-07-17 23:35:13
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        Schema::table('users', function (Blueprint $table): void {
+            $table->after('password', function (Blueprint $table): void {
+                $table->text('two_factor_secret')->nullable();
+                $table->text('two_factor_recovery_codes')->nullable();
+                $table->timestamp('two_factor_confirmed_at')->nullable();
+            });
+        });
+    }
+};
diff -ruN a/phpstan-baseline.neon b/phpstan-baseline.neon
--- a/phpstan-baseline.neon	2026-07-18 06:31:04
+++ b/phpstan-baseline.neon	2026-07-17 23:35:13
@@ -1,6 +1,39 @@
 parameters:
 	ignoreErrors:
 		-
+			message: '#^Parameter \#1 \$input \(array\<string, string\>\) of method App\\Actions\\Fortify\\CreateNewUser\:\:create\(\) should be contravariant with parameter \$input \(array\) of method Laravel\\Fortify\\Contracts\\CreatesNewUsers\:\:create\(\)$#'
+			identifier: method.childParameterType
+			count: 1
+			path: app/Actions/Fortify/CreateNewUser.php
+
+		-
+			message: '''
+				#^Return type of method App\\Actions\\Fortify\\CreateNewUser\:\:passwordRules\(\) has typehint with deprecated interface Illuminate\\Contracts\\Validation\\Rule\:
+				see ValidationRule$#
+			'''
+			identifier: return.deprecatedInterface
+			count: 1
+			path: app/Actions/Fortify/CreateNewUser.php
+
+		-
+			message: '''
+				#^Return type of method App\\Actions\\Fortify\\ResetUserPassword\:\:passwordRules\(\) has typehint with deprecated interface Illuminate\\Contracts\\Validation\\Rule\:
+				see ValidationRule$#
+			'''
+			identifier: return.deprecatedInterface
+			count: 1
+			path: app/Actions/Fortify/ResetUserPassword.php
+
+		-
+			message: '''
+				#^Return type of method App\\Actions\\Fortify\\UpdateUserPassword\:\:passwordRules\(\) has typehint with deprecated interface Illuminate\\Contracts\\Validation\\Rule\:
+				see ValidationRule$#
+			'''
+			identifier: return.deprecatedInterface
+			count: 1
+			path: app/Actions/Fortify/UpdateUserPassword.php
+
+		-
 			message: '#^Dynamic call to static method Illuminate\\Foundation\\Application\:\:isProduction\(\)\.$#'
 			identifier: staticMethod.dynamicCall
 			count: 2
diff -ruN a/routes/console.php b/routes/console.php
--- a/routes/console.php	2026-07-18 00:00:00
+++ b/routes/console.php	2026-07-18 00:00:00
@@ -5 +5,3 @@
 use Illuminate\Support\Facades\Schedule;
+
+Schedule::command('disposable:update')->weekly();
diff -ruN a/tests/Integration/Actions/UpdateUserProfileInformationTest.php b/tests/Integration/Actions/UpdateUserProfileInformationTest.php
--- a/tests/Integration/Actions/UpdateUserProfileInformationTest.php	1969-12-31 18:00:00
+++ b/tests/Integration/Actions/UpdateUserProfileInformationTest.php	2026-07-17 23:35:13
@@ -0,0 +1,104 @@
+<?php
+
+declare(strict_types=1);
+
+use App\Actions\Fortify\UpdateUserProfileInformation;
+use App\Models\User;
+use Illuminate\Auth\Notifications\VerifyEmail;
+use Illuminate\Support\Facades\Notification;
+use Illuminate\Support\Str;
+use Illuminate\Validation\ValidationException;
+
+it('validates fields', function (array $data, array $expected): void {
+    $user = User::factory()->createOne();
+
+    expect(fn () => resolve(UpdateUserProfileInformation::class)->update($user, $data))
+        ->toThrow(function (ValidationException $exception) use ($expected): void {
+            expect($exception->validator->errors()->messages())->toBe($expected);
+        });
+})->with([
+    'required' => [
+        'data' => [
+            'email' => '',
+            'first_name' => '',
+            'last_name' => '',
+        ],
+        'expected' => [
+            'email' => ['The email field is required.'],
+            'first_name' => ['The first name field is required.'],
+            'last_name' => ['The last name field is required.'],
+        ],
+    ],
+    'max:255 (string)' => [
+        'data' => [
+            'email' => 'valid@example.com',
+            'first_name' => Str::repeat('a', 256),
+            'last_name' => Str::repeat('a', 256),
+        ],
+        'expected' => [
+            'first_name' => ['The first name field must not be greater than 255 characters.'],
+            'last_name' => ['The last name field must not be greater than 255 characters.'],
+        ],
+    ],
+]);
+
+it('validates email uniqueness', function (): void {
+    User::factory()->createOne(['email' => 'taken@example.com']);
+
+    $user = User::factory()->createOne();
+
+    expect(fn () => resolve(UpdateUserProfileInformation::class)->update($user, [
+        'email' => 'taken@example.com',
+        'first_name' => 'Jane',
+        'last_name' => 'Doe',
+    ]))->toThrow(function (ValidationException $exception): void {
+        expect($exception->validator->errors()->messages()['email'])
+            ->toBe(['The email has already been taken.']);
+    });
+});
+
+it('updates the user profile name fields', function (): void {
+    $user = User::factory()->createOne([
+        'email' => 'old@example.com',
+        'email_verified_at' => now(),
+        'first_name' => 'Old',
+        'last_name' => 'Name',
+    ]);
+
+    resolve(UpdateUserProfileInformation::class)->update($user, [
+        'email' => 'old@example.com',
+        'first_name' => 'New',
+        'last_name' => 'Name',
+    ]);
+
+    expect($user->refresh())
+        ->email->toBe('old@example.com')
+        ->email_verified_at->not->toBeNull()
+        ->first_name->toBe('New')
+        ->last_name->toBe('Name');
+});
+
+it('marks the email as unverified when the email changes', function (): void {
+    Notification::fake();
+
+    $user = User::factory()->createOne([
+        'email' => 'old@example.com',
+        'email_verified_at' => now(),
+        'first_name' => 'Old',
+        'last_name' => 'Name',
+    ]);
+
+    resolve(UpdateUserProfileInformation::class)->update($user, [
+        'email' => 'new@example.com',
+        'first_name' => 'New',
+        'last_name' => 'Name',
+    ]);
+
+    expect($user->refresh())
+        ->email->toBe('new@example.com')
+        ->email_verified_at->toBeNull()
+        ->first_name->toBe('New')
+        ->last_name->toBe('Name');
+
+    Notification::assertSentTo($user, VerifyEmail::class);
+});
